src/detect-http-protocol.c

```diff
@@ -65,166 +65,77 @@
 
 #include "util-print.h"
 
-#define KEYWORD_NAME "http_protocol"
+#define KEYWORD_NAME "http.protocol"
```

Comment:
Please update https://github.com/OISF/suricata/blob/master/doc/userguide/rules/http-keywords.rst with the new name.